modules/container/container.go

```diff
@@ -0,0 +1,219 @@
+package container
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"syscall"
+	"time"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/oci"
+	"github.com/google/shlex"
+	"github.com/threefoldtech/zbus"
+	"github.com/threefoldtech/zosv2/modules"
+	"github.com/threefoldtech/zosv2/modules/stubs"
+)
+
+const (
+	containerdSock = "/run/containerd/containerd.sock"
+)
+
+type containerModule struct {
+	flister    *stubs.FlisterStub
+	containerd string
+}
+
+func New(zcl zbus.Client, containerd string) modules.ContainerModule {
+	if len(containerd) == 0 {
+		containerd = containerdSock
+	}
+
+	return &containerModule{
+		flister:    stubs.NewFlisterStub(zcl),
+		containerd: containerd,
+	}
+}
+
+func getNetworkSpec(network modules.NetworkInfo) oci.SpecOpts {
+	ns := network.Namespace
+	if !path.IsAbs(ns) {
+		// just name
+		ns = path.Join("/var/run/netns", ns)
+	}
+
+	return oci.WithLinuxNamespace(
+		specs.LinuxNamespace{
+			Type: specs.NetworkNamespace,
+			Path: ns,
+		},
+	)
+}
+
+func withHooks(hooks specs.Hooks) oci.SpecOpts {
+	return func(_ context.Context, _ oci.Client, _ *containers.Container, spec *oci.Spec) error {
+		spec.Hooks = &hooks
+		return nil
+	}
+}
+
+// NOTE:
+// THIS IS A WIP Create action and it's not fully implemented atm
+func (c *containerModule) Run(ns string, name string, flist string, tags, env []string, network modules.NetworkInfo,
+	mounts []modules.MountInfo, entrypoint string) (id modules.ContainerID, err error) {
+	// create a new client connected to the default socket path for containerd
+	client, err := containerd.New(c.containerd)
+	if err != nil {
+		return id, err
+	}
+	defer client.Close()
+
+	args, _ := shlex.Split(entrypoint)
```

Comment:
What does this return if `entrypoint` is empty ?

Reply:
it will return an empty []string list